tests/step-119.sh

```diff
@@ -0,0 +1,131 @@
+        git add . &&
+        git commit -a -m 'Testing 1' ||
+        return 1
+}
+
+makeShared 1 || exit 1
+makeShared 2 || exit 2
+makeShared 3 || exit 3
+makeShared 4 || exit 4
+
+# Install global hooks
+git config --global githooks.shared "file:///tmp/shared1.git,/tmp/shared2.git"
+
+# Make repo with local paths in .shared file
+mkdir -p /tmp/test119 && cd /tmp/test119 || exit 7
+git init &&
+    mkdir ".githooks" &&
+    touch ".githooks/trust-all" &&
+    echo "# Hooks A" >>.githooks/.shared &&
+    echo "file:///tmp/shared3.git" >>.githooks/.shared &&
+    echo "# Hooks B" >>.githooks/.shared &&
+    echo "/tmp/shared4.git" >>.githooks/.shared &&
+    git add . || exit 8
+
+git hooks config accept trusted || exit 9
+git hooks shared update || exit 10
+
+OUT=$(git commit -a -m "Testing" 2>&1)
+if ! echo "$OUT" | grep -q "Shared repo 1: pre-commit,Shared repo 2: pre-commit,Shared repo 3: pre-commit,Shared repo 4: pre-commit"; then
+    echo "! Expected to have run 4 shared hooks: $OUT" >&2
+    exit 10
+fi
+
+# Update to new version
+git config --global githooks.testingTreatFileProtocolAsRemote "true"
+
+cd ~/githooks-original &&
+    git reset --hard master-old &&
+    cp -r /var/lib/githooks/* ./ &&
+    git commit -a -m "Current Changes: Simulating a merge with this feature branch" || exit 11
+
+# Trigger update
+cd /tmp/test119 || exit 12
+UPDATE_OUT=$(git hooks update 2>&1)
+# shellcheck disable=SC2181
+if [ $? -ne 0 ]; then
+    echo "! Expected update to succeed: $UPDATE_OUT"
+    exit 1
+fi
+
+# Check if we have multiple githook.shared values now
+COUNT=$(git config --global --get-all githooks.shared | wc -l)
+if [ "$COUNT" != "2" ]; then
+    echo "! Expected 2 githooks.shared config variables after legacy transform"
+    git config --global --get-all githooks.shared
+    exit 1
+fi
+
+if echo "$EXTRA_INSTALL_ARGS" | grep -q "use-core-hookspath"; then
+    if ! echo "$UPDATE_OUT" | grep "DEPRECATION WARNING: Local paths for shared hook repositories"; then
+        echo "! Expected deprecation warning to move local paths manually: $UPDATE_OUT"
+        exit 1
+    fi
+
+    if [ "$(git config --global githooks.failOnNonExistingSharedHooks)" != "true" ]; then
+        echo "! Expected githooks.failOnNonExistingSharedHooks to be activated to help the user fix the problem"
+        exit 1
+    fi
+
+    # We are finished here, since we need to manually clean up...
+else
+
+    # Check if local path got moved to --local githooks.shared
+    if grep "shared4" /tmp/test119/.githooks/.shared ||
+        ! git config --local githooks.shared | grep -q "shared4"; then
+        echo "! Expected local path to be moved after legacy transform"
+        cat /tmp/test119/.githooks/.shared
+        echo "githooks.shared: $(git config --local githooks.shared)"
+        exit 1
+    fi
+
+    # We need a hooks update (folder naming has changed)
+    git hooks shared update || exit 13
```

Comment:
Don't you do this in the legacy transform already?
If not, I don't think it was clear from the messaging there that users will need to run this by themselves after the update.

Reply:
Correct I only (and we can only do this) in the core.hooksPath mode.
There is the warning:
```
 echo "! DEPRECATION WARNING: Because of renaming of internal cloned shared" >&2
            echo "  hook repositories, you should update all shared hook repositories" >&2
            echo "  by running in all repositories using Githooks:" >&2
            echo "    \$ git hooks shared update" >&2
```
in normal installation I trigger this directly: 
I should make this conditionally in the test on the install method... Lets try

Reply:
Hehe, another thing detected:

```
legacy_transform_adjust_local_paths() {

    if [ "$(git -C "$1" rev-parse --is-bare-repository 2>/dev/null)" = "true" ]; then
        SHARED_FILE="$1/.githooks/.shared"
    else
        SHARED_FILE="$1/../.githooks/.shared"
    fi
```
there is a new command `git worktree list --porcelain` availabe which **we strongly should use**. It works (with version > 2.25) also in a `.git` dir (also for bare repositories) and give a list of all worktrees attached -> the first one is the main worktree. When ever we want to find from the registered `.git` dir (--git-common-dir) the worktrees like here -> we should use this: I have to check where else we need this. Probably only in such circumstances where we need to do legacy transforms... 

So here we also need to loop through all worktree

Reply:
Works, see latest commit.